src/heretic/utils.py

```diff
@@ -36,7 +36,31 @@ def format_duration(seconds: float) -> str:
 
 
 def load_prompts(specification: DatasetSpecification) -> list[str]:
-    dataset = load_dataset(specification.dataset, split=specification.split)
+    import os
+
+    # Check if dataset path is a local directory
+    if os.path.isdir(specification.dataset):
+        # Load from local disk (Arrow format)
+        dataset_dict = load_from_disk(specification.dataset)
+        # Extract the split from the specification (e.g., "train[:400]")
+        split_name = specification.split.split("[")[0] if "[" in specification.split else specification.split
```

Comment:
```suggestion
        split_name = specification.split.split("[")[0]
```

I believe this can be simplified because IIRC, `str.split` returns a list containing the entire string if the string doesn't contain the separator.